scripts/data_collector/yahoo/collector.py

```diff
@@ -147,7 +148,29 @@ def _show_logging_func():
     def get_data(
         self, symbol: str, interval: str, start_datetime: pd.Timestamp, end_datetime: pd.Timestamp
     ) -> pd.DataFrame:
-        @deco_retry(retry_sleep=self.delay)
+        if hasattr(self, 'retry_config'):
```

Comment:
Will it look simpler  to make retry as a part of the class  interface ( for example, [abstract  attribute](https://stackoverflow.com/questions/2736255/abstract-attributes-in-python) )

Reply:
@you-n-g but should I define this class interface only for `YahooCollectorBR1d` and `YahooCollectorBR1min` classes?

Reply:
Maybe we can define

```
class YahooCollector(...):
      retry = 5  # Configuration attribute.  How many times will it try to re-request the data if the network fails.

class YahooCollectorBR(...):
        """"
            The reason to use retry=2 is due to the fact that
            Yahoo Finance unfortunately does not keep track of the majority
            of Brazilian stocks. 
            
            Therefore, the decorator deco_retry with retry argument
            set to 5 will keep trying to get the stock data 5 times, 
            which makes the code to download Brazilians stocks very slow. 
            
            In future, this may change, but for now 
            I suggest to leave retry argument to 1 or 2 in 
            order to improve download speed.
            In order to achieve this code logic an argument called retry_config
            was added into YahooCollectorBR base class
        """
      retry = 2
```

Now the retry is part of the interface, so we don't have to use hasattr to access the retry.

Reply:
@you-n-g I've added such modifications into commit https://github.com/microsoft/qlib/pull/990/commits/c31380481d189d5b7e05490dd9c662f975dffa97